### PR TITLE
Fix declaration name completion support around nullable-typed things

### PR DESCRIPTION
**Pull request by jasonmalinowski**
Looks like we never gave suggestions if the type name ended in a ?.

Fixes https://developercommunity2.visualstudio.com/t/Regression-from-1675-Suggested-varia/1220195

### CODE REVIEW

**Review by davidwengier — APPROVED** [review]

**Review by CyrusNajmabadi — APPROVED** [review]

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs [hunk 1]
```diff
@@ -1620,6 +1620,121 @@ public void Method()
             await VerifyItemExistsAsync(markup, "nullables");
         }
 
+        [WorkItem(1220195, "https://github.com/dotnet/roslyn/issues/23590")]
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs:1623]: @jasonmalinowski The work item link here doesn't seem correct.

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs [hunk 2]
```diff
@@ -1620,6 +1620,121 @@ public void Method()
...
+
+public class Class1
+{
+  public void Method()
+  {
+      Nullable<ImmutableArray<int>> $$
+  }
+}
+";
+            await VerifyItemExistsAsync(markup, "vs");
```
Comment (Youssef1313) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs:1640]: Related to: https://github.com/dotnet/roslyn/issues/36364#issuecomment-501812129

@sharwell What is the expected name here?

Comment (jasonmalinowski) [reply src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs:1640]: I'll happily chime in to say that while I may not have a strong opinion what we _should_ suggest, this suggestion is indeed confusing and nearly worthless.

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs (hunk 1 above)
Comment (jasonmalinowski) [reply src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs:1623]: Yeah, I totally didn't copy/paste that from the other test in the file and forget to update it. Totally.

:wink:

Fix here: https://github.com/dotnet/roslyn/pull/50246. Good catch!

**Pull request merged**